File: tests/sdk/go/helm-release/step1/main.go
package main

import (
	"fmt"

	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/helm/v3"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		var opts []pulumi.ResourceOption

		conf := config.New(ctx, "")
		chart := conf.Require("chart")
		version := conf.Require("version")
		var repoOpts *helm.RepositoryOptsArgs
		if repo, err := conf.Try("repo"); err == nil && repo != "" {
			repoOpts = &helm.RepositoryOptsArgs{
				Repo: pulumi.StringPtr(repo),
			}
		}
		values := map[string]interface{}{}
		conf.RequireObject("values", &values)

		rel, err := helm.NewRelease(ctx, "test", &helm.ReleaseArgs{
			Chart:          pulumi.String(chart),
			Version:        pulumi.String(version),
			RepositoryOpts: repoOpts,
			Values:         pulumi.ToMap(values),
			Timeout:        pulumi.Int(300),
		}, opts...)
		if err != nil {
			return err
		}

		// export the resourceNames for validation purposes
		ctx.Export("resourceNames", rel.ResourceNames)

		svc := pulumi.All(rel.Status.Namespace(), rel.Status.Name()).
			ApplyT(func(r any) (any, error) {
				arr := r.([]any)
				namespace := arr[0].(*string)
				name := arr[1].(*string)
				svc, err := corev1.GetService(ctx, "svc", pulumi.ID(fmt.Sprintf("%s/%s-nginx", *namespace, *name)), nil)
				if err != nil {
					return "", nil
				}
				return svc.Spec.ClusterIP(), nil
			})
		ctx.Export("svc_ip", svc)

		return nil
	})
}

File: tests/sdk/python/helm-release/step1/__main__.py
"""A Kubernetes Python Pulumi program"""

from pulumi_kubernetes.core.v1 import Namespace
from pulumi_kubernetes.helm.v3 import Release, ReleaseArgs, RepositoryOptsArgs

app_labels = {"app": "nginx"}

namespace = Namespace("test")

Release("release1",
        args=ReleaseArgs(
                chart="nginx",
                repository_opts=RepositoryOptsArgs(
                    repo="https://raw.githubusercontent.com/bitnami/charts/eb5f9a9513d987b519f0ecd732e7031241c50328/bitnami"
                ),
                namespace=namespace.metadata["name"],
                values={
                    "service": {"type": "ClusterIP"},
                    "image": {
                        "repository": "bitnamisecure",
                        "tag": "latest",
                    },
                },
                version="6.0.4",
            ))

# Deploy a duplicate chart release to verify that multiple instances of the Chart
# can be managed in the same stack.
Release("release2",
        args=ReleaseArgs(
                chart="nginx",
                repository_opts=RepositoryOptsArgs(
                    repo="https://raw.githubusercontent.com/bitnami/charts/eb5f9a9513d987b519f0ecd732e7031241c50328/bitnami"
                ),
                namespace=namespace.metadata["name"],
                values={
                    "service": {"type": "ClusterIP"},
                    "image": {
                        "repository": "bitnamisecure",
                        "tag": "latest",
                    },
                },
                version="6.0.4",
            ))
